queue_job/models/queue_job.py

```diff
@@ -324,19 +325,24 @@ def _change_job_state(self, state, result=None):
             else:
                 raise ValueError(f"State not supported: {state}")
 
-    def button_done(self):
+    def button_done(self, __state_in=(WAIT_DEPENDENCIES, PENDING, ENQUEUED, FAILED)):
```

Comment:
Why making `__state_in` a method argument?

Reply:
This dunder argument is a way to have constant defined once, when module is loaded. It will not be rebuilt on each method call.
Also when reading code it is easy to identify which state transitions are allowed. And prefix with dunder `__` makes it explicit that it is internal stuff.

If it is confusing, it can be moved to the 1st line of the method.

Reply:
> it can be moved to the 1st line of the method.

I would prefer that indeed.